### PR TITLE
Add flag to remove registry authentication

### PR DESCRIPTION
**Pull request by Shubham-astro**
## Description

This PR intends to remove the registry auth section using the flag to avoid mounting astronomer-tls to the registry pod.

## Related Issues

Do not merge this PR until this text is replaced with links to related issues.

Related astronomer/issues#6718

## Testing

Registry pod should be up and running without astronomer-tls certificate being created. 
```
(venv) shubhamsingh@Shubhams-MacBook-Pro astro % helm template astronomer --show-only charts/astronomer/templates/registry/registry-configmap.yaml . --set global.baseDomain=example.com --set astronomer.registry.enableInsecureAuth=True --debug
install.go:222: [debug] Original chart version: ""
install.go:239: [debug] CHART PATH: /Users/shubhamsingh/Desktop/git/astro

---
# Source: astronomer/charts/astronomer/templates/registry/registry-configmap.yaml
###################################
## Astronomer Registry ConfigMap ##
###################################

kind: ConfigMap
apiVersion: v1
metadata:
  name: astronomer-registry
  labels:
    component: registry
    tier: astronomer
    release: astronomer
    chart: "astronomer-0.36.0"
    heritage: Helm
  
data:
  config.yml: |-
    version: 0.1
    log:
      fields:
        service: registry
      level: info
    storage:
      cache:
        blobdescriptor: inmemory
      filesystem:
        rootdirectory: /var/lib/registry
      delete:
        enabled: true
    http:
      addr: :5000
      # debug:
      #   addr: :5001
      #   prometheus:
      #     enabled: true
      #     path: /metrics
      headers:
        X-Content-Type-Options: [nosniff]
    health:
      storagedriver:
        enabled: true
        interval: 10s
        threshold: 3
    notifications:
      endpoints:
        - name: "houston"
          url: "http://astronomer-houston:8871/v1/registry/events"
          timeout: 30s
          threshold: 10
          backoff: 1s
          ignoredmediatypes:
            - application/octet-stream
(venv) shubhamsingh@Shubhams-MacBook-Pro astro % 

```

## Merging

0.37

### CODE REVIEW

**Comment by pgvishnuram** [comment]
@Shubham-astro did you discuss this change with anybody

File: charts/astronomer/templates/registry/registry-configmap.yaml [hunk 1]
```diff
@@ -52,11 +52,15 @@ data:
       headers:
         X-Content-Type-Options: [nosniff]
     auth:
+      {{- if .Values.registry.enableInsecureReadingAndWriting }}
+      disabled: true # This disables authentication when the flag is set
```
Comment (danielhoherd) [inline charts/astronomer/templates/registry/registry-configmap.yaml:56]: IMHO the comment doesn't add anything that isn't obvious in the code.
```suggestion
      disabled: true
```

Comment (pgvishnuram) [reply charts/astronomer/templates/registry/registry-configmap.yaml:56]: this change has implications we cannot do this

Comment (danielhoherd) [reply charts/astronomer/templates/registry/registry-configmap.yaml:56]: We can't delete a comment?

File: charts/astronomer/values.yaml [hunk 2]
```diff
@@ -420,6 +420,8 @@ commander:
   extraVolumes: []
 
 registry:
+  # flag to disable astronomer registry
+  enableInsecureReadingAndWriting: False
```
Comment (rishkarajgi) [inline charts/astronomer/values.yaml:424]:
```suggestion
  enableInsecureAuth: False
```

File: charts/astronomer/values.yaml [hunk 3]
```diff
@@ -420,6 +420,8 @@ commander:
   extraVolumes: []
 
 registry:
+  # flag to disable astronomer registry
```
Comment (rishkarajgi) [inline charts/astronomer/values.yaml:423]:
```suggestion
  # flag to disable secure (tls) authorization to astronomer registry
```

**Review by rishkarajgi — APPROVED** [review]

File: charts/astronomer/templates/registry/registry-statefulset.yaml [hunk 4]
```diff
@@ -128,13 +130,15 @@ spec:
...
                 path: config.yml
+        {{- if not .Values.registry.enableInsecureAuth }}
         - name: certificate
           secret:
             secretName: {{ .Values.global.tlsSecret }}
         {{ include "custom_ca_volumes" . | indent 8  }}
         {{- if and .Values.registry.gcs.enabled .Values.registry.gcs.useKeyfile }}
         {{- include "registry.gcsVolume" . | indent 8 }}
         {{- end }}
+        {{- end }}
```
Comment (pgvishnuram) [inline charts/astronomer/templates/registry/registry-statefulset.yaml:141]: on Line no 89 - 92 you have added logic to remove only ssl certs but here in this condition you are removing everything

Comment (Shubham-astro) [reply charts/astronomer/templates/registry/registry-statefulset.yaml]: corrected.

**Review by pgvishnuram — APPROVED** [review]

**Review by rishkarajgi — APPROVED** [review]

**Pull request merged**